internal/validator/default/helpers.go

```diff
@@ -58,6 +60,77 @@ func extendTx(ctx context.Context, w validator.TxFinderI, rawTx *bt.Tx) error {
 	return nil
 }
 
+// getUnminedAncestors returns unmined ancestors with data necessary to perform Deep Fee validation
+func getUnminedAncestors(ctx context.Context, w validator.TxFinderI, tx *bt.Tx) (map[string]*bt.Tx, error) {
+	unmindedAncestorsSet := make(map[string]*bt.Tx)
+
+	// get distinct parents
+	// map parentID with inputs collection to avoid duplication and simplify later processing
+	parentInputMap := make(map[string][]*bt.Input)
+	parentsIDs := make([]string, 0, len(tx.Inputs))
+
+	for _, in := range tx.Inputs {
+		prevTxID := in.PreviousTxIDStr()
+
+		inputs, found := parentInputMap[prevTxID]
+		if !found {
+			// first occurrence of the parent
+			inputs = make([]*bt.Input, 0)
+			parentsIDs = append(parentsIDs, prevTxID)
+		}
+
+		inputs = append(inputs, in)
+		parentInputMap[prevTxID] = inputs
+	}
+
+	// get parents
+	const finderSource = validator.SourceTransactionHandler | validator.SourceWoC
+	parentsTxs, err := w.GetRawTxs(ctx, finderSource, parentsIDs)
+	if err != nil {
+		return nil, fmt.Errorf("cannot extend tx: %w", err)
+	}
+
+	if len(parentsTxs) != len(parentsIDs) {
+		return nil, errParentNotFound
+	}
+
+	for _, p := range parentsTxs {
+		if _, found := unmindedAncestorsSet[p.TxID]; found {
+			continue // parent was proccesed already
+		}
+
+		childInputs, found := parentInputMap[p.TxID]
+		if !found {
+			return nil, errParentNotFound
+		}
+
+		// fulfill data about the parent for furhter validation
```

Comment:
I assume you didn't want to say "furher" 😄 
```suggestion
		// fulfill data about the parent for further validation
```

Reply:
fixed